src/core/src/op/shape_of.cpp

```diff
@@ -2,93 +2,33 @@
-}
-
-void op::v3::ShapeOf::validate_and_infer_types() {
-    OV_OP_SCOPE(v3_ShapeOf_validate_and_infer_types);
-    NODE_VALIDATION_CHECK(this,
-                          m_output_type == element::i64 || m_output_type == element::i32,
-                          "Output type must be i32 or i64");
-    set_input_is_relevant_to_value(0, false);
-    const auto input_partial_shape = get_input_partial_shape(0);
-    set_output_type(0, m_output_type, ov::PartialShape{input_partial_shape.rank()});
-}
-
-bool ngraph::op::v3::ShapeOf::visit_attributes(AttributeVisitor& visitor) {
-    OV_OP_SCOPE(v3_ShapeOf_visit_attributes);
-    visitor.on_attribute("output_type", m_output_type);
-    return true;
-}
-
-shared_ptr<Node> op::v3::ShapeOf::clone_with_new_inputs(const OutputVector& new_args) const {
-    OV_OP_SCOPE(v3_ShapeOf_clone_with_new_inputs);
-    check_new_args_count(this, new_args);
-    auto new_shape_of = make_shared<op::v3::ShapeOf>(new_args.at(0), m_output_type);
-    return new_shape_of;
-}
-
-OPENVINO_SUPPRESS_DEPRECATED_START
+namespace ov {
+namespace op {
 namespace shape_of {
 namespace {
 template <element::Type_t ET>
-inline bool evaluate(const ov::Shape& shape, const HostTensorPtr& output_value) {
-    ov::reference::shape_of(shape, output_value->get_data_ptr<ET>());
-    return true;
-}
-
-template <element::Type_t ET>
-inline bool evaluate(const ov::Shape& shape, ov::Tensor& output_value) {
-    ov::reference::shape_of(shape, output_value.data<fundamental_type_for<ET>>());
+inline bool evaluate(const Shape& shape, Tensor& output_value) {
+    reference::shape_of(shape, output_value.data<fundamental_type_for<ET>>());
     return true;
 }
 
-bool evaluate_shape_of(const HostTensorPtr& output_value, const HostTensorPtr& input_value) {
-    bool rc = true;
-    ov::Shape shape = input_value->get_shape();
-    output_value->set_shape(ov::Shape{shape.size()});
-    switch (output_value->get_element_type()) {
-        OPENVINO_TYPE_CASE(evaluate_shape_of, i32, shape, output_value);
-        OPENVINO_TYPE_CASE(evaluate_shape_of, i64, shape, output_value);
-        OPENVINO_TYPE_CASE(evaluate_shape_of, u32, shape, output_value);
-        OPENVINO_TYPE_CASE(evaluate_shape_of, u64, shape, output_value);
-    default:
-        rc = false;
-        break;
-    }
-    return rc;
-}
-
-bool evaluate_shape_of(ov::Tensor& output_value, const Shape& input_shape) {
+bool evaluate_shape_of(Tensor& output_value, const Shape& input_shape) {
     bool rc;
-    output_value.set_shape(ov::Shape{input_shape.size()});
+    output_value.set_shape(Shape{input_shape.size()});
     switch (output_value.get_element_type()) {
         OPENVINO_TYPE_CASE(evaluate_shape_of, i32, input_shape, output_value);
         OPENVINO_TYPE_CASE(evaluate_shape_of, i64, input_shape, output_value);
-        OPENVINO_TYPE_CASE(evaluate_shape_of, u32, input_shape, output_value);
-        OPENVINO_TYPE_CASE(evaluate_shape_of, u64, input_shape, output_value);
```

Comment:
I'm not sure if this can be removed, @jane-intel  could you comment if this can or cannot be removed?

Reply:
I suggest to move such significant changes into separate PRs, so as not to complicate the transition to the new API.

Reply:
Reverted